quinn-udp/src/unix.rs

```diff
@@ -20,24 +20,48 @@ type IpTosTy = libc::c_uchar;
 #[cfg(not(target_os = "freebsd"))]
 type IpTosTy = libc::c_int;
 
+trait StdUdpSocketExt {
+    fn only_v6(&self) -> io::Result<bool>;
+}
+
+impl StdUdpSocketExt for std::net::UdpSocket {
+    fn only_v6(&self) -> io::Result<bool> {
+        let raw_fd = self.as_raw_fd();
+        let mut val: libc::c_int = 0;
+        let mut len = mem::size_of::<libc::c_int>() as libc::socklen_t;
+        let res = unsafe {
+            libc::getsockopt(
+                raw_fd,
+                libc::IPPROTO_IPV6,
+                libc::IPV6_V6ONLY,
+                &mut val as *mut _ as *mut _,
+                &mut len,
+            )
+        };
+        if res == -1 {
```

Comment:
Style nit: I think this would be nicer as a `match` expression:

```rust
match res {
    -1 => Err(io::Error::last_os_err()),
    _ => Ok(val != 0),
}
```

Reply:
will do